app/common/helpers/collections.py

```diff
@@ -59,6 +66,14 @@ def name(self) -> str:
     def reference(self) -> str:
         return self.submission.reference
 
+    @property
+    def expression_context(self) -> ExpressionContext:
+        return ExpressionContext(
+            from_submission=immutabledict(
+                {mangle_question_id_for_context(uuid.UUID(k)): v for k, v in self.submission.data.items()}
```

Comment:
currently this fetches straight out of the submission data, do we want to start it off with the contexts being guarded by question types? I'm still thinking through some of the implications (in a comment above) but we could do that tidily for the submission helper without deeping performance too hard

```suggestion
                {mangle_question_id_for_context(uuid.UUID(key)): self.get_answer_for_question(key).root for key in self.submission.data.keys() }
```